### PR TITLE
Initialize CVPipelineEditor with last stage selected and stage grid focused

### PR DESCRIPTION
**Pull request by JulienDevillers**
# Description
Initialize CVPipelineEditor with last stage selected and stage grid focused

# Justification
User see immediately the result of the pipeline.
He can immediately navigate through stages using up/down arrow keys.

# Instructions for Use
Open any pipeline editor, press up/down keys.

This is documentation for a user, not for a developer.

# Implementation Details
1. How did you test the change? Be descriptive. Untested code will not be accepted.
Manual tests.
Mini refactored the actual code to remove redundant parametering of the dialog.

2. Did you follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)?
Yes ! love them.

3. If you made changes in the `org.openpnp.spi` or `org.openpnp.model` packages you will need to add additional justification for these changes. Changes to these packages require extensive review and testing.
No

4. Be sure to run `mvn test` before submitting the Pull Request. If the tests do not pass the Pull Request will not be accepted.

### CODE REVIEW

**Comment by markmaker** [comment]
Hi Julien

Nice to have this initialization in a central place. For instance now the initial window size could be managed (store/restore or make a percentage of the main frame). So the window is not so unnecesary small on a large monitor. 

Coming from C++ I'm a bit unsure whether it is good to do all this initialization in the constructor. Maybe a static method would be more appropriate? I guess @vonnieda needs to take a look.

More questions in the code.

_m

**Comment by JulienDevillers** [comment]
Hi Mark,

My primary goal was UX. Refactoring is a side effect... :)

What do you mean by 'more question in the codes ?

File: src/main/java/org/openpnp/machine/reference/vision/wizards/ReferenceFiducialLocatorPartConfigurationWizard.java [hunk 1]
```diff
@@ -92,12 +92,11 @@ private void editPipeline() throws Exception {
         pipeline.setProperty("package", part.getPackage());
         pipeline.setProperty("footprint", part.getPackage().getFootprint());
         CvPipelineEditor editor = new CvPipelineEditor(pipeline);
-        JDialog dialog = new JDialog(MainFrame.get(), "Fiducial Locator Pipeline");
-        dialog.getContentPane().setLayout(new BorderLayout());
-        dialog.getContentPane().add(editor);
-        dialog.setSize(1024, 768);
+        JDialog dialog = new CvPipelineEditorDialog(MainFrame.get(), "Fiducial Locator Pipeline", editor);
         dialog.setVisible(true);
+        editor.initializeFocus();
```
Comment (markmaker) [inline src/main/java/org/openpnp/machine/reference/vision/wizards/ReferenceFiducialLocatorPartConfigurationWizard.java]: Shouldn't this line be removed?

Comment (JulienDevillers) [reply src/main/java/org/openpnp/machine/reference/vision/wizards/ReferenceFiducialLocatorPartConfigurationWizard.java]: Hi Mark,
You're right, it's done.

**Comment by vonnieda** [comment]
Looks good to me, and a nice cleanup. Thank you!

 @markmaker, in Java doing this kind of initialization in the constructor is more a matter of style than anything. There aren't the potential pitfalls as there are in C++. Doing it in a static method would also be fine, but either is good.

**Pull request merged**